Dashboard/app/js/lib/views/surveys/question-view.js

```diff
@@ -197,6 +206,16 @@ FLOW.QuestionView = FLOW.View.extend({
     	FLOW.selectedControl.set('selectedCascadeResource', cascadeResource);
     }
 
+    // reset selected caddisfly resource
+    FLOW.selectedControl.set('selectedCaddisflyResource', null);
+    // if the caddisflyResourceUuid is not null, get the resource
+    if (!Ember.empty(FLOW.selectedControl.selectedQuestion.get('caddisflyResourceUuid'))) {
+      FLOW.caddisflyResourceControl.get('content').forEach(function (item) {
+        if (item.get('uuid') == FLOW.selectedControl.selectedQuestion.get('caddisflyResourceUuid')) {
+          FLOW.selectedControl.set('selectedCaddisflyResource', item);
```

Comment:
Try: 
`FLOW.caddisflyResourceControl.get('content').findProperty('uuid', FLOW.selectedControl.selectedQuestion.get('caddisflyResourceUuid'))`

>    Returns the first item with a property matching the passed value.  You
>    can pass an optional second argument with the target value.  Otherwise
>    this will match any property that evaluates to true.
>    This method works much like the more generic find() method.

https://github.com/akvo/akvo-flow/blob/v1.9.10/Dashboard/app/js/vendor/ember-1.0.0.pre-2-36.js#L6954-L6967